LamestWebserver/LamestWebserver/Core/Logger.cs

```diff
@@ -259,28 +341,68 @@ public void CrashAndBurn(string msg)
             Environment.Exit(-1);
         }
 
-        internal void Log(ELoggingLevel loggingLevel, string msg)
+        private void Log(ELoggingLevel loggingLevel, string msg)
         {
-            // TODO: Implement propper logging!
-
-            ServerHandler.LogMessage($"[{loggingLevel.ToString()} \\\\\\\\ {msg}]");
+            if (_currentOutputSource != EOutputSource.None)
+            {
+                using (_loggerWriteLock.LockWrite())
+                {
```

Comment:
`if (_multiStreamWriter !=null && !_multiStreamWriter.IsDisposed)`
if something like that exists - immediately makes `Close` & `Open` thread-safe, at the risk of dropping a few messages.
I'd prefer thread-safety for everything over not-dropping some packets if someone is really trying to manually call `Restart`. just add a description to `Close` and `Open` saying: `If you want to restart the stream manually use 'Restart'.`